### PR TITLE
MESOS: Move suicide logic into its own file

### PR DESCRIPTION
**Pull request by sttts**
Move suicide code from main scheduler.go into its own file to, simplifying the scheduler.

### CODE REVIEW

File: contrib/mesos/pkg/executor/suicide.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	log "github.com/golang/glog"
+	bindings "github.com/mesos/mesos-go/executor"
+)
+
+// TODO(jdef) add metrics for this?
+type suicideTimer struct {
+	timer *time.Timer
+}
+
+func (w *suicideTimer) Next(d time.Duration, driver bindings.ExecutorDriver, f jumper) suicideWatcher {
```
Comment (jdef) [inline contrib/mesos/pkg/executor/suicide.go]: probably want to move the declaration of the `jumper` type into this file too

Comment (jdef) [inline contrib/mesos/pkg/executor/suicide.go]: same for `suicideWatcher`

Comment (jdef) [inline contrib/mesos/pkg/executor/suicide.go]: there's probably also some unit tests that could go into a suicide_test.go file

**Comment by sttts** [comment]
@jdef addressed all your comments. ptal.

**Comment by jdef** [comment]
lgtm

**Comment by jdef** [comment]
@sttts why was this force-pushed? rebase?

**Comment by sttts** [comment]
Removed two unused imports in executor_test.go which caused Shippable to fail.

**Comment by sttts** [comment]
And another try with Shippable. Please! You can make it!

**Comment by sttts** [comment]
green. amen.

**Pull request merged**